### DISCUSSION TITLE
Explain CAN data (from CANdump logfile) through .dbc file?

### DISCUSSION THREAD

**Discussion by klausz** [cantools/cantools#582] [Q&A] [answered]
This command works very well:
candump vcan0 | cantools decode --single-line '/home/bla/bla2/xyz_cp.dbc' to get the CAN bus data mapped/translated. 

If and how is this possible with a CANdump file and .dbc instead of CANplayer and vcan0? 

Thats in the CANdump file. 
_(1689845934.809000) can0 00000101#4f021102000000a7
(1689845934.811000) can0 00000102#00002000000000cb_
..

I tried with Jupyter notebooks but unfortunately there is the error message a CANdump.log cannot be used. 
Even removing the .log gave an error message:
"_ValueError: expected database format 'arxml', 'dbc', 'kcd', 'sym', 'cdd' or None, but got_ ''"

As alternative i was thinking of Wireshark. But there is nothing.

**Comment by andlaus** [comment, marked as answer]
I have some trouble understanding what you mean, but if you want to decode a file containing can trace more directly, you can use
```
cat my-cantrace.log | cantools decode --single-line '/home/bla/bla2/xyz_cp.dbc'
```

where `my-cantrace.log` was created using, e.g.
```
candump -L can0 | tee mycantrace.log
```

**Reply by klausz** [reply]
Just tested in Jupyter:
`cat my-cantrace.log | cantools decode --single-line '/home/bla/bla2/xyz_cp.dbc'`
Using an existing logfile, the .dbc file for one exiting node and as result i get the 'dbc translated' CAN data. 
Just removed the '--single-line' 
`cat my-cantrace.log | cantools decode '/home/bla/bla2/xyz_cp.dbc `
It works well. Thank you. 
I will drill down to specific entries/messages. 
To visualize, there are examples.
